client/src/pages/SignUp.tsx

```diff
@@ -1,28 +1,31 @@
-import { useCreateUserWithEmailAndPassword } from 'react-firebase-hooks/auth';
+import { useCreateUserWithEmailAndPassword, useUpdateProfile } from 'react-firebase-hooks/auth';
 import { auth } from '../firebase/connection';
 import { useNavigate } from 'react-router-dom';
-import classes from './Sign.module.scss';
 import Form, { FormOptions } from '../components/Form';
+import Header from '../components/Header';
 
 const SignUpPage = () => {
   const navigate = useNavigate();
-  const [createUserWithEmailAndPassword, user] = useCreateUserWithEmailAndPassword(auth);
+  const [createUserWithEmailAndPassword] = useCreateUserWithEmailAndPassword(auth);
+  const [updateProfile] = useUpdateProfile(auth);
 
-  const handleFormSubmit = ({ email, password }: FormOptions) => {
-    createUserWithEmailAndPassword(email as string, password as string);
+  const handleFormSubmit = async ({ name, email, password }: FormOptions) => {
+    console.log('here');
+    
+    const user = await createUserWithEmailAndPassword(email as string, password as string);
+    if (user) {
+      const success = await updateProfile({ displayName: name });
+      if (success) {
+        navigate('/');
+      }
+    }
   };
 
-  if (user) {
-    navigate('/');
-  }
-
```

Comment:
This was used to redirect people who are already signed in.
This is also relevant to the sign in page.